doc/source/how-to-use-driver-api.rst

```diff
@@ -0,0 +1,254 @@
+
+    # Create a RecordSet
+    recordset = RecordSet()
+
+    # Add model parameters to the RecordSet
+    recordset.parameters_records["my_model"] = pytorch_to_parameter_record(global_model)
+
+    # Add a training configuration for 1 epoch only to the RecordSet
+    recordset.configs_records["my_config"] = ConfigsRecord({"epochs": 1})
+
+Next, we create a list of :code:`Messages`, one for each node ID. To do so, we loop over all node IDs and run :code:`Driver.create_message()` with the :code:`recordset` as the content of the message: 
+
+.. code-block:: python
+
+    # File: server.py
+
+    messages = []
+    for node_id in node_ids:
+        message = driver.create_message(
+            content=recordset,
+            message_type=MessageType.TRAIN,
+            dst_node_id=node_id,
+            group_id=str(server_round),
+            ttl=DEFAULT_TTL,
+        )
+        messages.append(message)
+
+Finally, we use :code:`Driver.push_messages()` to push the list of :code:`Messages` containing the encapsulated parameters and configurations to the nodes.
+
+.. code-block:: python
+
+    # File: server.py
+
+    message_ids = driver.push_messages(messages)
+
+:code:`Driver.push_messages()` yields an iterable list of message IDs. In some real-world scenarios, you may encounter situations where only some :code:`Messages` can be pushed, so it is good practice to filter out empty message IDs:
+
+.. code-block:: python
+
+    # File: server.py
+
+    # Wait for results, ignore empty message_ids
+    message_ids = [message_id for message_id in message_ids if message_id != ""]
+
+
+Pull messages from nodes
+------------------------
+
+Once messages are successfully sent to the nodes, we can use the associated message IDs to get results from these nodes. To do so, we continuously run :code:`Driver.pull_messages()` with the list of :code:`message_ids` until all of the :code:`Messages` from the nodes are received. 
+
+.. code-block:: python
+
+    # File: server.py
+
+    all_replies: List[Message] = []
+    while True:
+        replies = driver.pull_messages(message_ids=message_ids)
+        for res in replies:
+            print(f"Got 1 {'result' if res.has_content() else 'error'}")
+        all_replies += replies
+        if len(all_replies) == len(message_ids):
+            break
+        print("Pulling messages...")
+        time.sleep(3)
+
+To only keep :code:`Messages` with content, we apply a simple filter on the results:
+
+.. code-block:: python
+
+    # File: server.py
+
+    # Filter correct results
+    all_replies = [
+        msg
+        for msg in all_replies
+        if msg.has_content()
+    ]
+    print(f"Received {len(all_replies)} results")
+
+
+Process results from nodes
+--------------------------
+
+Now that we have a results from the nodes in the form of :code:`Messages`, we can access their content and use them for any subsequent server-side tasks. Here is how we print the :code:`metrics_records` for each node in a for-loop:
```

Comment:
similarly to my previous comment, should we stick to `Messages` (with that `s`) or change it?

Reply:
I've updated them to `Message` objects.